addons/product/models/product_template.py

```diff
@@ -967,24 +1023,24 @@ def _get_possible_combinations(self, parent_combination=None, necessary_values=N
 
         necessary_values = necessary_values or self.env['product.template.attribute.value']
         necessary_attributes = necessary_values.mapped('attribute_id')
-        ptal_stack = [self.valid_product_template_attribute_line_ids.filtered(lambda ptal: ptal.attribute_id not in necessary_attributes)]
-        combination_stack = [necessary_values]
-
-        # keep going while we have attribute lines to test
-        while len(ptal_stack):
-            attribute_lines = ptal_stack.pop()
-            combination = combination_stack.pop()
-
-            if not attribute_lines:
-                # full combination, if it's possible return it, otherwise skip it
-                if self._is_combination_possible(combination, parent_combination):
-                    yield(combination)
-            else:
-                # we have remaining attribute lines to consider
-                for ptav in reversed(attribute_lines[0].product_template_value_ids):
-                    ptal_stack.append(attribute_lines[1:])
-                    combination_stack.append(combination + ptav)
 
+        attributes = self.valid_product_template_attribute_line_ids.filtered(lambda ptal: ptal.attribute_id not in necessary_attributes)
```

Comment:
Now that I see it I think this line is actually bugged in v13 for multiple lines for the same attribute. It should be filtered on "necessary_values.attribute_line_id" and not on generic "attribute_id".

Reply:
Let's take an example to check I understand correctly.
Let's say we have an attribute `color = [red, green, blue]`.
You mean that a product table can have two attributes lines `legs color` and `top color` and that currently, it only searches one value by attribute rather than attribute line, correct ?

Well, I changed code to use `attribute_line_id` and it passes tests.

Reply:
In v13 you can have the following attribute lines on a product at the same time:
- color: red or green
- color: blue

Combinations are "red & blue" and "green & blue".
If you pass "color blue" as "necessary" this would incorrectly exclude the other color line "color: red" and fail to find anything.
There is no test covering this case so you can add it too.

This scenario is not possible in v12 (attribute line is restricted to 1 per attribute), but the fix can be made here as well.

Reply:
I think we need to fix this in 13.0, not 12.0 : 
```
AttributeError: 'product.template.attribute.line' object has no attribute 'attribute_line_id' 
```

Reply:
Good catch, for some reason I forgot the distinction was not possible in v12 (even though it was the main reason to refactor it in the first place). We should adapt that part from v13 then.

Reply:
But you probably also didn't call it on the right element, `attribute_line_id` should be called on the value (ptav), not on the line itself.

Reply:
I tested both
```
AttributeError: 'product.template.attribute.value' object has no attribute 'attribute_line_id'
```